source/util/bit_vector.h

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef LIBSPIRV_UTILS_BIT_VECTOR_H_
+#define LIBSPIRV_UTILS_BIT_VECTOR_H_
+
+#include <cassert>
+#include <iostream>
+#include <vector>
+
+#include "ilist_node.h"
+
+namespace spvtools {
+namespace utils {
+
+// Implements a bit vector class.
+//
+// All bits default to zero, and the upper bound is 2^32-1.
+class BitVector {
+ public:
+  // Creates a bit vector contianing 0s.
+  BitVector() : bits(1024 / sizeof(BitContainer), 0) {}
+
+  // Sets the |i|th bit to 1.
+  void Set(uint32_t i) {
+    uint32_t element_index = i / sizeof(BitContainer);
```

Comment:
sizeof returns the number of bytes. Should be dividing by 64 here to get the right vector element. The 63rd bit is in index 0, not index 7.